### PR TITLE
[#156465908] feat: added mailup configuration (dns + credentials)

### PR DESCRIPTION
**Pull request by gunzip**
While we can safely apply this configuration to Terraform (production) without any impact,
we must keep the sendgrid api key until we publish a new release to the production Functions
(which will cause APIs breaking changes)

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by cloudify** [comment]
if we keep the sendgrid key in the live system, pls remove that change from this PR, we'll change it later when we can remove it from the live system

File: infrastructure/azure.tf [hunk 1]
```diff
@@ -160,6 +160,23 @@ variable "SENDGRID_KEY" {
   description = "The API key for the SendGrid service"
 }
 
+# This should be passed bya ENV var TF_VAR_MAILUP_USERNAME
```
Comment (cloudify) [inline infrastructure/azure.tf]: typo

File: infrastructure/azure.tf [hunk 2]
```diff
@@ -160,6 +160,23 @@ variable "SENDGRID_KEY" {
...
+variable "MAILUP_USERNAME" {
+  type        = "string"
+  description = "Username for the MailUp SMTP+ service"
+}
+
+# This should be passed bya ENV var TF_VAR_MAILUP_SECRET
```
Comment (cloudify) [inline infrastructure/azure.tf]: typo

File: infrastructure/azure.tf [hunk 3]
```diff
@@ -845,7 +880,7 @@ resource "null_resource" "azurerm_apim_api" {
   }
 }
 
-# DNS Zone
+## DNS Zone
```
Comment (cloudify) [inline infrastructure/azure.tf]: to mark a "block", use the following format (this is not markdown):

```
#
# DNS Zone
#
```

File: infrastructure/azure.tf [hunk 4]
```diff
@@ -855,7 +890,27 @@ resource "azurerm_dns_zone" "azurerm_dns_main_zone" {
   count = "${var.environment == "production" ? 1 : 0}"
 }
 
-# Azure Container Service (Kubernetes)
+# default email provider txt spf1 dns record
```
Comment (cloudify) [inline infrastructure/azure.tf]: @gvarisco @paolo-de-rosa can you do a quick check?

File: infrastructure/azure.tf [hunk 5]
```diff
@@ -855,7 +890,27 @@ resource "azurerm_dns_zone" "azurerm_dns_main_zone" {
...
+resource "azurerm_dns_cname_record" "azurerm_dns_main_zone_dkim" {
+  count               = "${var.environment == "production" ? length(keys(var.azurerm_dns_main_zone_dkim)) : 0}"
+  resource_group_name = "${azurerm_resource_group.azurerm_resource_group.name}"
+  zone_name           = "${azurerm_dns_zone.azurerm_dns_main_zone.name}"
+  name                = "${element(keys(var.azurerm_dns_main_zone_dkim), count.index)}"
+  record              = "${lookup(var.azurerm_dns_main_zone_dkim, element(keys(var.azurerm_dns_main_zone_dkim), count.index))}"
+  ttl                 = 3600
+}
+
+## Azure Container Service (Kubernetes)
```
Comment (cloudify) [inline infrastructure/azure.tf]: ditto

File: infrastructure/env/common/tfvars.json [hunk 6]
```diff
@@ -15,5 +15,11 @@
   "azurerm_apim_eventhub_rule": "apim-eventhub-rule",
   "azurerm_kubernetes_admin_username": "k8s-admin",
   "azurerm_kubernetes_admin_ssh_publickey_file": "k8s_admin_key.pub",
-  "azurerm_dns_main_zone": "io.italia.it"
+  "default_sender_email": "no-reply@io.italia.it",
+  "azurerm_dns_main_zone": "io.italia.it",
+  "azurerm_dns_main_zone_spf1": "v=spf1 include:musv.com ~all",
```
Comment (rasky) [inline infrastructure/env/common/tfvars.json]: I don't understand what this setting is supposed to mean.

Comment (gunzip) [reply infrastructure/env/common/tfvars.json]: this setups a dns zone (server) to handle the io.italia.it subdomain. then it adds a txt record for spf1 and two cnames for dkim. the default_sender_email param is the default email address we use to send any email from d.c.

**Comment by gunzip** [comment]
I've checked the stats on the production api gateway. As there are very few requests in the last month I suppose it's safe to relase the lastest version of the backend API to production so we can migrate to mailup and get rid of sendgrid @cloudify

**Pull request merged**